src/scrolling/scrolling.service.ts

```diff
@@ -1,41 +1,57 @@
 import { Injectable } from '@nestjs/common';
 import { InjectModel } from '@nestjs/mongoose';
-import { error, table } from 'console';
 import { Model } from 'mongoose';
+import { ObjectId } from 'mongodb';
 import { Pet } from './schema/pet.schema';
+import { matchAnimals } from '../utils/matchAnimals';
+import { User, UserDocument } from '../auth/schemas/user.schema';
 
 @Injectable()
 export class ScrollingService {
-    constructor(
-        @InjectModel('Pet') private readonly petModel: Model<Pet>,
-    ) {}
-
-
-    async getPetbyIndex(id: string): Promise<Pet | { error: string }> {
-        const index = parseInt(id);
-        if (isNaN(index) || index < 0 ) {
-            return { error: 'Index is Invalid.' };
-          }
-        const result = await this.petModel.findOne({ id: index }).exec();
-          if(!result){
-            return {error: 'Pet not found.'}
-          }
-        return result;
-        
+  constructor(
+    @InjectModel('Pet') private readonly petModel: Model<Pet>,
+    @InjectModel(User.name) private userModel: Model<UserDocument>,
```

Comment:
please keep injecting consistent. 'Pet' is injected directly while User is being injected by model's name. Keeping it up to standard is crucial, because it makes your code more predictable. Please use one of:
```suggestion
    @InjectModel(Pet.name) private readonly petModel: Model<Pet>,
    @InjectModel(User.name) private userModel: Model<UserDocument>,
```
which I personally prefer
or 
```suggestion
    @InjectModel('Pet') private readonly petModel: Model<Pet>,
    @InjectModel('User') private userModel: Model<UserDocument>,
```